tests/base/src/test/java/org/wildfly/security/sasl/test/SaslAuthenticationTimeoutTest.java

```diff
@@ -0,0 +1,201 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.sasl.test;
+
+import static java.security.AccessController.doPrivileged;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.PrivilegedAction;
+import java.security.Provider;
+import java.security.Security;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.sasl.Sasl;
+import javax.security.sasl.SaslClient;
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+
+import org.jboss.threads.JBossThreadFactory;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.security.auth.client.AuthenticationConfiguration;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client.ClientUtils;
+import org.wildfly.security.auth.client.MatchRule;
+import org.wildfly.security.password.Password;
+import org.wildfly.security.password.PasswordFactory;
+import org.wildfly.security.password.WildFlyElytronPasswordProvider;
+import org.wildfly.security.password.interfaces.ClearPassword;
+import org.wildfly.security.password.spec.ClearPasswordSpec;
+import org.wildfly.security.sasl.SaslMechanismSelector;
+import org.wildfly.security.sasl.digest.DigestServerFactory;
+import org.wildfly.security.sasl.digest.WildFlyElytronSaslDigestProvider;
+import org.wildfly.security.sasl.util.AuthenticationTimeoutSaslServerFactory;
+import org.wildfly.security.sasl.util.SaslMechanismInformation;
+
+import mockit.Mock;
+import mockit.MockUp;
+
+/**
+ * Tests a successful authentication timeout for a custom executor service and the default executor service.
+ *
+ * @author <a href="mailto:aabdelsa@redhat.com">Ashley Abdel-Sayed</a>
+ */
+
+public class SaslAuthenticationTimeoutTest {
+
+    private static final String DIGEST = SaslMechanismInformation.Names.DIGEST_MD5;
+    private static final String AUTHENTICATION_TIMEOUT_MESSAGE = "Authentication mechanism server timed out";
+
+    private static final Provider[] providers = new Provider[] {
+            WildFlyElytronSaslDigestProvider.getInstance(),
+            WildFlyElytronPasswordProvider.getInstance()
+    };
+
+    /*
+     * Unable to set custom AUTHENTICATION_TIMEOUT using a property SaslServer factory (see ELY-1815), so using mock function
```

Comment:
Just minor, it would be good to mention here that the this method is being mocked to avoid using the DEFAULT_TIMEOUT of 150s.